### PR TITLE
Support `std::unique_ptr`.

### PR DESCRIPTION
**Pull request by zoecarver**

### CODE REVIEW

**Comment by zoecarver** [comment]
@swift-ci please smoke test

**Comment by zoecarver** [comment]
This is the tests from #65577

**Comment by hyp** [comment]
Thanks, having tests in tree is great!

**Review by hyp — APPROVED** [review]

File: test/Interop/Cxx/stdlib/std-unique-ptr.swift [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+// REQUIRES: executable_test
+
+import StdlibUnittest
+import StdUniquePtr
+#if os(Linux)
+import CxxStdlib
+// FIXME: import CxxStdlib.string once libstdc++ is split into submodules.
+#else
+import CxxStdlib.memory
+#endif
```
Comment (egorzhdan) [inline test/Interop/Cxx/stdlib/std-unique-ptr.swift:12]: Could you please change this to `import CxxStdlib` unconditionally?

**Pull request closed without merging**